test/indexer/test_hca_indexer.py

```diff
@@ -138,6 +143,10 @@ def tearDown(self):
     translated_bool_true = null_bool.to_index(True)
     translated_bool_false = null_bool.to_index(False)
 
+    @lru_cache(maxsize=None)
```

Comment:
It looks like you are uncertain when to use @cached_property and when to use @lru_cache. Feel free to ask after stand-up.